messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/InfiniteOrPositiveValidators.java

```diff
@@ -51,10 +60,24 @@ public void validateParameter(String parameterName, ModelNode value) throws Oper
         public void validateParameter(String parameterName, ModelNode value) throws OperationFailedException {
             super.validateParameter(parameterName, value);
             int val = value.asInt();
-            if (!(val == -1 || (val > 0 && val < Integer.MAX_VALUE))) {
+            if (!(val <= -1 || (val > 0 && val < Integer.MAX_VALUE))) {
                 throw new OperationFailedException(MessagingLogger.ROOT_LOGGER.illegalValue(value, parameterName));
             }
         }
     };
 
+    /**
+     * Correct any negative value to use -1 (interpreted by Artemis as infinite).
+     */
+    ParameterCorrector NEGATIVE_VALUE_CORRECTOR = new ParameterCorrector() {
+        @Override
+        public ModelNode correct(ModelNode newValue, ModelNode currentValue) {
+            if (newValue.isDefined() && newValue.getType() != ModelType.EXPRESSION && newValue.asLong() < -1) {
```

Comment:
@jmesnil This should catch any RuntimeException (i.e. from newValue.asLong() if that conversion is not possible) and just return newValue. That way the execution will continue to the validation part and the validator will see the 'wrong' value and can provide the proper failure for a wrong value.

Really, AttributeDefinition should handle that kind of safety check itself internally but it's late to do that for WF 12.

Reply:
I amended my commit to fix it. thanks